File: retrofit/src/main/java/com/kcst/retrofit/BaseResponse.java
package com.kcst.retrofit;

public class BaseResponse<T> {
    public boolean isSuccess;

    public int errorCode;

    public String errorMsg;

    public T data;


    public boolean isSuccess() {
        return isSuccess;
    }

    public void setSuccess(boolean success) {
        isSuccess = success;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "BaseResponse{" +
                "isSuccess=" + isSuccess +
                ", errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                '}';
    }
}

File: retrofit/src/main/java/com/kcst/retrofit/BaseResponse.kt
package com.kcst.retrofit

open abstract class BaseResponse<T> {
    @JvmField
    var isSuccess = false
    @JvmField
    var errorCode = 0
    @JvmField
    var errorMsg: String? = null

    abstract var data: T
    override fun toString(): String {
        return "BaseResponse(isSuccess=$isSuccess, errorCode=$errorCode, errorMsg=$errorMsg, data=$data)"
    }


}